src/main/java/models/Group.java

```diff
@@ -0,0 +1,50 @@
+package models;
+
+import java.util.Objects;
+
+public class Group {
+    private int id;
+    private String name;
+
+    public Group (String name) {
+        this.name = name;
+    }
+
+    public Group (int id, String name) {
+        this.id = id;
+        this.name = name;
+    }
+
+    public void setId(int id) {
+        this.id = id;
+    }
+
+    public int getId() {
+        return id;
+    }
+
+    public String getName() {
+        return name;
+    }
+
+    @Override
+    public boolean equals(Object o) {
+        if (this == o) return true;
+        if (o == null || getClass() != o.getClass()) return false;
+        Group group = (Group) o;
+        return id == group.id && Objects.equals(name, group.name);
```

Comment:
pls use refers to the current object this
And it's seems that student.id also called id from current object. Because you can't get it from other object without get method